src/main/scala/it/pagopa/interop/apigateway/service/impl/NotifierServiceImpl.scala

```diff
@@ -0,0 +1,48 @@
+package it.pagopa.interop.apigateway.service.impl
+
+import it.pagopa.interop.apigateway.service.{NotifierInvoker, NotifierService}
+import it.pagopa.interop.commons.utils.errors.GenericComponentErrors
+import it.pagopa.interop.notifier.client.api.EventsApi
+import it.pagopa.interop.notifier.client.invoker.BearerToken
+import it.pagopa.interop.notifier.client.model.Messages
+import it.pagopa.interop.partymanagement.client.invoker.ApiError
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class NotifierServiceImpl(invoker: NotifierInvoker, api: EventsApi)(implicit ec: ExecutionContext)
+    extends NotifierService {
+
+  implicit val logger: Logger = LoggerFactory.getLogger(this.getClass)
+
+  override def getEvents(lastEventId: String, limit: Int)(contexts: Seq[(String, String)]): Future[Messages] = {
+    for {
+      (bearerToken, correlationId, ip) <- extractHeadersWithOptionalCorrelationIdF(contexts)
```

Comment:
FYI: this will be 
```suggestion
      (bearerToken, correlationId, ip) <- extractHeaders(contexts).toFuture
```
like [here](https://github.com/pagopa/interop-be-api-gateway/pull/35/files#diff-e41846488afb3862163f13a83edbfcf47e92f539beb3e02fbac7b6339d8969edL21) (once that PR will be merged)